src/dune_pkg/fetch.ml

```diff
@@ -12,30 +17,80 @@ module Curl = struct
        | Some v -> base ^ "." ^ Build_info.V1.Version.to_string v)
   ;;
 
+  module Fiber_lazy : sig
+    type 'a t
+
+    val create : (unit -> 'a Fiber.t) -> 'a t
+    val force : 'a t -> 'a Fiber.t
+  end = struct
+    type 'a t =
+      { value : 'a Fiber.Ivar.t
+      ; mutable f : (unit -> 'a Fiber.t) option
+      }
+
+    let create f = { f = Some f; value = Fiber.Ivar.create () }
+
+    let force t =
+      match t.f with
+      | None -> Fiber.Ivar.read t.value
+      | Some f ->
+        Fiber.of_thunk (fun () ->
+          t.f <- None;
+          let* v = f () in
+          let+ () = Fiber.Ivar.fill t.value v in
+          v)
+    ;;
+  end
+
+  let compressed_supported =
+    (* We check if curl supports --compressed by running curl -V and checking if
+       the output contains the features we need. *)
+    Fiber_lazy.create (fun () ->
+      let+ lines, _ =
+        let stderr_to = Process.Io.null Out in
+        Process.run_capture_lines
+          Return
+          ~stderr_to
+          ~display:Quiet
+          (Lazy.force bin)
+          [ "-V" ]
+      in
+      (* If these features are present, then --compressed is supported *)
+      let features = [ "libz"; "brotli"; "zstd" ] in
+      match lines with
+      | [ _; _; _; curl_features ] ->
```

Comment:
Careful, this will break if curl adds or removes any of these lines (e.g. the user has a git version of curl that might not have the second line which contains the release date), I'd say this is a bit too fragile.

I'd suggest a `find_map` with `seq [bol; str "Features:"; …]` instead.